### DISCUSSION TITLE
VS Code Extension

### DISCUSSION THREAD

**Discussion by janosh** [Authenticator-Extension/Authenticator#1080] [Ideas]
A lot of remote `ssh` sessions these days require 2FA. Would be great if there was a way to auto-fill the OTP when connecting via VS Code Remote.

![Screenshot 2023-07-18 at 3 49 41 PM](https://github.com/Authenticator-Extension/Authenticator/assets/30958850/28715992-bc45-4e7f-b4b7-d542202c14b1)

**Comment by mymindstorm** [comment]
We don't have much control of anything outside of the browser unfortunately. Your best bet here would be an SSH key or certificate based auth. If you really want to, it should be possible to get the functionality you want via either some sort of vscode extension or a script that runs during ssh authentication.

**Reply by janosh** [reply]
Thanks for the quick reply. Not sure I understand though. Afaik VS Code extensions can run any TypeScript code and interact with any part of the editor UI. What kind of control is lacking?

**Reply by mymindstorm** [reply]
First, try following this guide https://www.cyberciti.biz/faq/how-to-set-up-ssh-keys-on-linux-unix/. If that does not work, then consider hacking something together.
